tests/integration/input/test_number_input.py

```diff
@@ -0,0 +1,63 @@
+import sys
+from selenium.webdriver.common.keys import Keys
+
+
+def test_inni001_invalid_numbers(ninput_app, dash_duo):
+    dash_duo.start_server(ninput_app)
+    for invalid_number in (
+        "10e10000",
+        "e+++eeeeeE-",
+        "120.2.33",
+        "12-.3",
+        "-122.33-",
+        "-3.3.3",
+        "0..0",
+    ):
+        for debounce in ("false", "true"):
+
+            elem = dash_duo.find_element("#input_{}".format(debounce))
+            assert not elem.get_attribute("value")
+
+            elem.send_keys(invalid_number)
+            elem.send_keys(Keys.TAB)  # onblur
+
+            # the number input element is special now, selenium doesn't work
+            dash_duo.percy_snapshot("inni001 - invalid number input onblur")
```

Comment:
Unable to look at the Percy results atm but from the style above, assuming these invalid values do not highlight the input red.

Reply:
needs a fix on dash.testing side for the percy issue, will work on that. 

the current behavior is close to native one, you get red on firefox though